sstables/data_consume_context.hh

```diff
@@ -124,8 +125,7 @@ public:
     ~data_consume_context() {
         if (_ctx) {
             auto f = _ctx->close();
-            //FIXME: discarded future.
-            (void)f.handle_exception([ctx = std::move(_ctx), sst = std::move(_sst)](auto) {});
+            (void)f.handle_exception([ctx = std::move(_ctx), sst = std::move(_sst), op = db::start_background_job()](auto) {});
```

Comment:
What about moving this to sstable_manager? It already tracks the background close of the sstable itself.

Reply:
I imagined that we will want to track all background operations, not just those which originate in the sstable layer. So the mechanism shouldn't be sstable specific. 

But rather, we should add db::start_background_job() to `sstable_manager::deactivate(sstable*)`.

Reply:
Different users will have different meanings for "all". Of course we have a single user now, so it's not yet a problem.

What about approaching it from another direction? @bhalevy is working on adding close() to readers, so they won't go out of scope in normal operation.

Reply:
(not go out of scope while having work to do)

Reply:
There is at least one user which needs "all" to mean every background operation which is triggered by the database layer. We can have more fine-grained tracking at different layers, but we will still have to combine them into the meaning expected by tests. We can split the tracking later if necessary.

Once @bhalevy's work goes in, it could replace some of these, but not all. It won't cover the sstable manager part, for example.

Reply:
Bu the sstable manager part won't be hit by this test, so this patch will become unnecessary.